Lib/asyncio/base_events.py

```diff
@@ -1170,6 +1170,9 @@ async def create_connection(
                         # the various error messages.
                         raise OSError('Multiple exceptions: {}'.format(
                             ', '.join(str(exc) for exc in exceptions)))
+                    else:
+                        # No exceptions were collected, raise a generic connection error
+                        raise OSError('create_connection failed')
```

Comment:
Would not be better to raise a `TimeoutError`? This only happens when all connections exceed their time.